dotnet/src/Connectors/Connectors.OpenAI/Files/OpenAIFilePurpose.cs

```diff
@@ -8,15 +9,90 @@ namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+    /// <summary>
+    /// File produced as result of a file included as a batch request.
+    /// </summary>
+    public static OpenAIFilePurpose BatchOutput { get; } = new("batch_output");
+
+    /// <summary>
+    /// File to be used as input to fine-tune a model.
+    /// </summary>
+    public static OpenAIFilePurpose FineTune { get; } = new("fine-tune");
+
+    /// <summary>
+    /// File produced as result of fine-tuning a model.
+    /// </summary>
+    public static OpenAIFilePurpose FineTuneResults { get; } = new("fine-tune-results");
+
+    /// <summary>
+    /// File to be used for Assistants image file inputs.
+    /// </summary>
+    public static OpenAIFilePurpose Vision { get; } = new("vision");
+
+    /// <summary>
+    /// Gets the label associated with this OpenAIFilePurpose.
+    /// </summary>
+    public string Label { get; }
+
+    /// <summary>
+    /// Creates a new OpenAIFilePurpose instance with the provided label.
+    /// </summary>
+    /// <param name="label">The label to associate with this AuthorRole.</param>
+    public OpenAIFilePurpose(string label)
+    {
+        Verify.NotNullOrWhiteSpace(label, nameof(label));
+        this.Label = label!;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether two AuthorRole instances are equivalent, as determined by a
+    /// case-insensitive comparison of their labels.
+    /// </summary>
+    /// <param name="left"> the first AuthorRole instance to compare </param>
+    /// <param name="right"> the second AuthorRole instance to compare </param>
+    /// <returns> true if left and right are both null or have equivalent labels; false otherwise </returns>
+    public static bool operator ==(OpenAIFilePurpose left, OpenAIFilePurpose right)
+        => left.Equals(right);
+
+    /// <summary>
+    /// Returns a value indicating whether two AuthorRole instances are not equivalent, as determined by a
+    /// case-insensitive comparison of their labels.
+    /// </summary>
+    /// <param name="left"> the first AuthorRole instance to compare </param>
+    /// <param name="right"> the second AuthorRole instance to compare </param>
+    /// <returns> false if left and right are both null or have equivalent labels; true otherwise </returns>
+    public static bool operator !=(OpenAIFilePurpose left, OpenAIFilePurpose right)
+        => !(left == right);
+
+    /// <inheritdoc/>
+    public override bool Equals([NotNullWhen(true)] object? obj)
+        => obj is OpenAIFilePurpose otherRole && this == otherRole;
+
+    /// <inheritdoc/>
+    public bool Equals(OpenAIFilePurpose other)
+        => string.Equals(this.Label, other.Label, StringComparison.OrdinalIgnoreCase);
+
+    /// <inheritdoc/>
+    public override int GetHashCode()
+        => StringComparer.OrdinalIgnoreCase.GetHashCode(this.Label ?? string.Empty);
+
+    /// <inheritdoc/>
+    public override string ToString() => this.Label ?? string.Empty;
```

Comment:
nit: Label is declared as not nullable and additionality there's a null guard in the constructor, so the "?? string.Empty" piece can be dropped.

Reply:
I noticed this also, but err'ed on the side of retaining alignment with AuthorRole.  Updated.